tfjs-layers/src/layers/preprocessing/preprocessing_utils.ts

```diff
@@ -0,0 +1,84 @@
+import { Tensor, denseBincount, Tensor1D, Tensor2D, TensorLike, mul} from '@tensorflow/tfjs-core';
+import { getExactlyOneTensor } from '../../utils/types_utils';
+import { expandDims} from '@tensorflow/tfjs-core';
+import { ValueError } from '../../errors';
+import * as K from '../../backend/tfjs_backend';
+
+export const int = 'int';
+export const oneHot = 'oneHot';
+export const multiHot = 'multiHot';
+export const count = 'count';
+export const tfIdf = 'tfIdf';
+
+export function encodeCategoricalInputs(inputs: Tensor|Tensor[],
+                                        outputMode: string,
+                                        depth: number,
+                                        weights?: Tensor1D|Tensor2D|TensorLike):
+                                        Tensor|Tensor[] {
+
+  inputs = getExactlyOneTensor(inputs);
+
+  if(inputs.dtype !== 'int32') {
+        inputs = K.cast(inputs, 'int32');
+    }
+
+  if(outputMode === int) {
+    return inputs;
+  }
```

Comment:
I don't see `int` output mode ever passed to this function, and I can't find it in the [Keras docs for category encoding](https://www.tensorflow.org/api_docs/python/tf/keras/layers/CategoryEncoding). What is it for?